### ISSUE TITLE
Drop support for Python 3.3

### ISSUE THREAD

**Issue by hugovk** [SoCo/SoCo#525]
## Reasons for dropping

Python 3.3 went [EOL last month](https://en.wikipedia.org/wiki/CPython#Version_history) and is no longer receiving security updates.

* Here's the pip installs for SoCo from PyPI for the last month,  showing low numbers for 3.3:
```
$ pypinfo --percent --pip soco pyversion
python_version percent download_count
-------------- ------- --------------
3.5              37.2%            823
3.6              29.0%            641
2.7              17.0%            377
3.4              15.3%            338
3.3               1.5%             33
2.6               0.0%              1
```
* pip 10 will deprecate Python 3.3 support, pip 11 won't support it (https://github.com/pypa/pip/issues/3796)
* Very little PyPI traffic (June 2016) https://github.com/pypa/pip/issues/3796 and even lower pip installs for the last month:
```
$ pypinfo --percent --pip "" pyversion
python_version percent download_count
-------------- ------- --------------
2.7              76.8%    488,818,554
3.6               9.9%     62,911,206
3.5               7.9%     50,014,064
3.4               4.0%     25,283,462
2.6               1.2%      7,848,827
3.3               0.1%        755,895
3.7               0.1%        400,727
3.2               0.0%         51,588
None              0.0%         19,451
2.8               0.0%            202
3.1               0.0%            169
```
* Allows the introduction of non-3.3 code without the concern of 3.3 failing
* Frees up resources on Travis CI for SoCo and everyone else

## Reasons for keeping

* Supporting 3.3 in addition to 2.7/3.4+ isn't as difficult as supporting 2.6 or 3.2

---

Anything else, for or against?

An alternative to dropping is deprecating: announcing it will be removed in a future version.

**Comment by stefankoegl** [comment]
+1 for dropping 3.3 after 0.13 is released (#442).

**Comment by KennethNielsen** [comment]
-1 I don't think we need  to push dropping support before either:

 * We are held back from using newer features. (If this is the case, I would like an example).
 * We can't run the tests

Since SoCo is (almost) always run on peoples private network, I think it is safe to say that security considerations are minimal and their peoples own to think about and possibly ignore.

**Comment by hugovk** [comment]
Here's what's new in 3.4: https://docs.python.org/3.6/whatsnew/3.4.html

asyncio, enum and pathlib perhaps the most useful, but I guess you'll need to support 2.7 anyway until 2020 or so.

**Comment by KennethNielsen** [comment]
@hugovk oh yeah, I know about those, but yes, we will keep 2.7 around for a while, so the interesting question is, which features are 3.3 keeping us from, that is available in 2.7 (possibly as a backported PyPI package) and 3.3+, that we would like to use.

Regarding what's new in 3.4 I'm not overly enthusiastic for SoCo; enum is a curiosity to satisfy low level programmers who can't bring themselves to use a dict and its inverted. asyncio is, just as all other async libraries that we might consider using a more complicated programming model, for advantages that I have yet to see demonstrated is relevant for SoCo, pathlib does sound awesome for many projects, but perhaps not so much for SoCo.
